src/satosa/micro_services/account_linking.py

```diff
@@ -52,6 +53,10 @@ def _handle_al_response(self, context):
             satosa_logging(logger, logging.INFO, "issuer/id pair is linked in AL service",
                            context.state)
             internal_response.user_id = message
+            id_to_attr = self.config.get("id_to_attr", None)
+            if id_to_attr:
+                internal_response.attributes[id_to_attr] = message
```

Comment:
Which brings me to the next point, which is that attributes are stored as an array of attribute-values. I think this should be

```python
internal_response.attributes[id_to_attr] = [message]
#                                          ^ notice the brackets
```